tests/repositories/test_legacy_repository.py

```diff
@@ -309,16 +309,16 @@ def test_get_200_returns_page(http):
     assert repo._get("/foo")
 
 
-def test_get_404_returns_none(http):
-    repo = MockHttpRepository({"/foo": 404}, http)
+def test_get_401_403_404_and_returns_none(http):
```

Comment:
```suggestion
def test_get_40x_and_returns_none(http):
```

Additioanlly would be great to parameterise it instead of using a single test case.